apps/prosody/templates/volumes/storage.yml.j2

```
@@ -0,0 +1,18 @@
+{% if prosody_storage == "internal" %}
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: prosody-pvc-storage
+  namespace: "{{ namespace_name }}"
+  labels:
+    app: prosody
+    version: "{{ prosody_image_tag }}"
+    deployment_stamp: "{{ deployment_stamp }}"
+spec:
+  accessModes:
+  - ReadWriteMany
+  resources:
+    requests:
+      storage: {{ prosody_storage_volume_size }}
+  storageClassName: "{{ default_storage_class_rwx }}"
```

```suggestion
  - ReadWriteOnce
  resources:
    requests:
      storage: {{ prosody_storage_volume_size }}
  storageClassName: "{{ default_storage_class_rwo }}"
```

Since there is only one instance of prosody at the same time (replicas:1 + non blue green app), I would suggest to use a `ReadWriteOnce` persistent volume.